packages/tracing/tests/helpers/tracesUtils.ts

```diff
@@ -117,6 +122,28 @@ const getTraces = async (xrayClient: XRay, startTime: Date, resourceArn: string,
       })).sort((a, b) => a.Document.start_time - b.Document.start_time) as ParsedSegment[],
     })).sort((a, b) => a.Segments[0].Document.start_time - b.Segments[0].Document.start_time);
 
+    // Verify that all trace has fully loaded invocation subsegments.
+    // The subsegments may be not available yet or still in progress.
+    for (const trace of sortedTraces) {
+      let retryFlag = false;
+      
+      let invocationSubsegment;
+      try {
+        invocationSubsegment = getInvocationSubsegment(trace);
+      } catch (error) {
+        if (error instanceof FunctionSegmentNotDefinedError){
+          retry(new Error(`There is no Function subsegment (AWS::Lambda::Function) yet. Retry.`));
+        } else {
+          throw error;
+        }
+      }
+
+      retryFlag = retryFlag || (!!invocationSubsegment.in_progress);
```

Comment:
Question: essentially here we are looking at the `in_progress` key. If it's set, we retry. Correct?

Reply:
I see! thanks for clarifying